lib/mix/lib/mix/tasks/test.ex

```diff
@@ -17,14 +19,64 @@ defmodule Mix.Tasks.Test do
       output = opts[:output]
 
       fn ->
-        Mix.shell().info("\nGenerating cover results ...")
         File.mkdir_p!(output)
 
-        Enum.each(:cover.modules(), fn mod ->
-          {:ok, _} = :cover.analyse_to_file(mod, '#{output}/#{mod}.html', [:html])
-        end)
+        Mix.shell().info("\nCover results")
+
+        {:result, ok, _fail} = :cover.analyse(:coverage, :module)
+
+        Mix.shell().info("Percentage | Module")
+        Mix.shell().info("-----------|--------------------------")
+
+        total =
+          ok
+          |> Stream.each(&display(&1, opts))
+          |> Stream.each(&html(&1, output))
+          |> Enum.reduce({0, 0}, fn {_, {cov, not_cov}}, {tot_cov, tot_not_cov} ->
+            {tot_cov + cov, tot_not_cov + not_cov}
+          end)
+
+        Mix.shell().info("-----------|--------------------------")
+
+        display({"Total", total}, opts)
       end
     end
+
+    defp colour(percentage, threshold) when percentage > threshold, do: :green
+    defp colour(_, _), do: :red
+
+    defp display({name, coverage}, opts) do
+      threshold = Keyword.get(opts, :threshold, @threshold)
+      percentage = percentage(coverage)
+
+      Mix.shell().info([
+        colour(percentage, threshold),
+        format(percentage, 9),
+        "%",
+        :reset,
+        " | #{name}"
```

Comment:
You need to inspect the module name so it shows only as "MyApp.Module" and not "Elixir.MyApp.Module". But you need to be careful to not inspect the string "Total".

Reply:
That is the reason why I haven't done it, however I can make it happen.